### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by DesignrKnight**
### Your checklist for this pull request
🚨Please review the [guidelines for contributing](../CONTRIBUTING.md) to this repository.

- [ ] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). Don't request your default branch!
- [ ] Make sure you are making a pull request against the **default branch** (left side). Also you should start *your branch* off *default branch*.
- [ ] Check the commit's or even all commits' message styles matches our requested structure.
- [ ] Check your code additions will fail neither code linting checks nor unit test.
- [ ] I have added necessary documentation (if appropriate)

### Description
Please describe your pull request.

💔Thank you!

### Post merge checklist
- [ ] Follow steps from the [guidelines for contributing](../CONTRIBUTING.md) to this repository.
- [ ] If you are a new contributor, ping in the thread and one of the maintainers will add you to all-contributors list.

### CODE REVIEW

**Comment by DesignrKnight** [comment]
This is draft until work is finished. Do not merge until then

**Comment by DesignrKnight** [comment]
WIP by @haru-02 and @Priya-81199

**Comment by DesignrKnight** [comment]
@Chinmay-KB @Thesmader 
Can you do final check and merge. Also, do add them to the all-contributors. 
Create an issue to add them to App's Contributor's page

Wonderful work @Priya-81199 and @haru-02

**Comment by Priya-81199** [comment]
> @Chinmay-KB @Thesmader
> Can you do final check and merge. Also, do add them to the all-contributors.
> Create an issue to add them to App's Contributor's page
> 
> Wonderful work @Priya-81199 and @haru-02

Thank you Abel😇

**Comment by haru-02** [comment]
> @Chinmay-KB @Thesmader
> Can you do final check and merge. Also, do add them to the all-contributors.
> Create an issue to add them to App's Contributor's page
> 
> Wonderful work @Priya-81199 and @haru-02

Thank you bhaiya 😄

**Review by Chinmay-KB — APPROVED** [review]

**Comment by DesignrKnight** [comment]
@all-contributors add @priya for code, maintenance

**Comment by DesignrKnight** [comment]
@all-contributors add @Priya-81199  for code, maintenance

**Comment by DesignrKnight** [comment]
@all-contributors add @haru-02 for Code, maintenance

File: scp/lib/services/graphQLservice.dart [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    HttpLink(uri: "https://ics.dscnitrourkela.org/graphql");
+
+class GraphQLService{
+
+ValueNotifier<GraphQLClient> valueclient = ValueNotifier<GraphQLClient>(
+    GraphQLClient(
+        link: httpLink,
+        cache: OptimisticCache(dataIdFromObject: typenameDataIdFromObject)));
+
+}
```
Comment (Thesmader) [inline scp/lib/services/graphQLservice.dart:14]:
```suggestion
}

```

**Review by Thesmader — APPROVED** [review]

**Pull request merged**